packages/store-ui/src/atoms/List/List.tsx

```diff
@@ -0,0 +1,59 @@
+import type { HTMLAttributes, ElementType } from 'react'
+import React, { forwardRef } from 'react'
+
+export type ListVariant = 'default' | 'description' | 'ordered' | 'unordered'
+
+const variantToType = (variant: string) => {
+  switch (variant) {
+    case 'description':
+      return 'dl'
+
+    case 'unordered':
+      return 'ul'
+
+    case 'ordered':
+      return 'ol'
+
+    default:
+      return 'div'
+  }
+}
+
+export interface ListProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * Set the HTML element tag of this component.
+   */
+  as?: ElementType
+  /**
+   * ID to find this component in testing tools (e.g.: cypress, testing library, and jest).
+   */
+  testId?: string
+  /**
+   * The current use case variant for Lists.
+   */
+  variant?: ListVariant
+}
+
+export const List = forwardRef<HTMLDivElement, ListProps>(function List(
+  {
+    children,
+    testId = 'store-list',
+    variant = 'default',
+    as: Component = variantToType(variant),
+    ...rawProps
+  },
+  ref
+) {
+  const props = {
+    [`data-store-list-${variant}`]: true,
```

Comment:
We will discuss it.

Reply:
I think there should be two props for this:

```tsx
data-store-list: true
data-${variant}: true
```

I was discussing this matter with @emersonlaurentino and we couldn't get to an agreement on when to use an attribute value (eg. `'data-store-list': variant`) or a different data attribute as I proposed. I'd love to hear @igorbrasileiro @victorhmp inputs on this.

Emerson suggested for us to use different data attributes only when dealing with state variables, such as `loading`, `error`, `success`, etc. Although I can kinda feel a conceptual difference between these variables and other variables such as `variaton`, we were not able to verbally define the concept, which makes me think they are the same, and thus shouldn't be treated differently.

Also, I think data attributes without values with data attributes with values can confuse users on when to use what. For me, if we have to explain the concept and the mindset behind it, we've already lost the game. We should adopt one or the other. For me, I pick different `data-attributes` for each value. What do you all think?

Reply:
Yeah, this is a tough one 😅 .

I think we should think about how to make it easier for users to compose their CSS rules. The first use case that came into mind was of a user trying to apply CSS rules to **all** `List` components, and then separate rules for each list variation. If we were to use a single data attribute for that, users would be able to achieve this by doing:

```css
[data-store-list] {
  /* all lists */
}

[data-store-list="unordered"] {
  /* only unordered lists */
}
```

If we go the other way and add two props, from the user's perspective, they would achieve the same with these selectors:

```css
[data-store-list] {
  /* all lists */
}

[data-store-list][data-unordered] {
  /* only unordered lists */
}
```

Honestly, I don't think any of them is more complex than the other one, so I would be fine with both. But, for the sake of consistency, I think @icazevedo's suggestion is more intuitive. Because then all of our data-attributes would always be `boolean` values and that's very predictable. 

I also had the same impression that different data attributes would only be a good idea for _state_, that's how I used them in the Carousel for example. But I guess you could argue that adding a specific selector for a given state would also make that a variant... 🤔

Reply:
According to the `reach.tech` [docs](https://reach.tech/styling/#pseudo-selectors) we should use the second way.

```ts
const props = {
   [`data-${variant}`]: true,
}
```

I see a pros using this approach, if we have the same name for data-variant cross components, we can share CSS through the data-attribute.

Reply:
Yes, but they also contradict themselves by using both approaches on [their components](https://reach.tech/accordion) 😆 

```css
[data-reach-accordion-item] {
  /* styles for all accordion items */
}
[data-reach-accordion-item][data-state="open"] {
  /* styles for all open accordion items */
}
[data-reach-accordion-item][data-state="collapsed"] {
  /* styles for all collapsed accordion items */
}
[data-reach-accordion-item][data-disabled] {
  /* styles for all disabled accordion items */
}
[data-reach-accordion-item][data-read-only] {
  /* styles for all read-only accordion items */
}
```

Reply:
So we need to discuss it better.

Reply:
@icazevedo I wouldn't say they're contradicting themselves, but they are doing the opposite of what we thought about hahahaha they use two props for variations and these variable values for state. I think we should go one step further and treat these two things as the same thing. In our case, everything being a variation (resulting in always using boolean data attributes)

Reply:
Btw, I updated the code in this branch according to this idea, so you can see how it looks 😄

Reply:
I liked a lot how they're handling variant/state. I think it makes sense to use a `data-state` that changes it value according to which is the current state, and it also is good to have one data attribute for each variation. In our case we are only dealing with variants in this PR, so the way @victorhmp updated now is perfect for me.

Reply:
Yeah, I prefer the current approach too. It makes more consistent when we are using `data-{variant}`, making sure it'll be a boolean value. We will open too much possibilities for values if we put a string value inside the `data-state` and people will be able to overwrite this if they want, making it a little bit non-standard.